src/modules/objects/KvsObject_painter.cpp

```diff
@@ -754,15 +754,75 @@ KVSO_END_CONSTRUCTOR(KvsObject_painter)
 		}                                                                                                                    \
 	}
 
+#define KVSO_6OR4ARRAYPARAMETERS(__pXOrArray,__iY,__iW,__iH,__iA1,__iA2)                                                                    \
+	if(__pXOrArray->isArray())                                                                                                              \
+	{                                                                                                                                       \
+		if(__pXOrArray->array()->size() < 4)                                                                                                \
+		{                                                                                                                                   \
+			c->error(__tr2qs_ctx("The array passed as parameter must contain at least 4 elements","objects"));                              \
+			return false;                                                                                                                   \
+		}                                                                                                                                   \
+		if(__pXOrArray->array()->size() < 6)                                                                                                \
+		{                                                                                                                                   \
+			if(c->params()->count() < 3)                                                                                                    \
+			{                                                                                                                               \
+				QString error=function+__tr2qs_ctx(" requires either an array as first parameter and 2 integers or six integers","objects");\
```

Comment:
`__tr2qs_ctx` is for translation of message into other languages.
You are incorrectly assuming that subject of sentence (in this case, name of function) is at the beginning of sentence in all languages.

Reply:
I know how to use __tr2qs_ctx I was just trying to stick as close as possible to the style of the rest of the file, but you are right.